sound/soc/sof/debug-dsp-ops.c

```diff
@@ -213,5 +245,9 @@ int sof_dbg_dsp_ops_test_init(struct snd_sof_dev *sdev)
 	if (ret < 0)
 		return ret;
 
-	return sof_dsp_dsp_ops_create_dfse(sdev, "boot_fw", dsp_ops_debugfs, 0222);
+	ret = sof_dsp_dsp_ops_create_dfse(sdev, "boot_fw", dsp_ops_debugfs, 0222);
+	if (ret < 0)
+		return ret;
+
+	return sof_dsp_dsp_ops_create_dfse(sdev, "dsp_power_state", dsp_ops_debugfs, 0666);
```

Comment:
The problem I have with this API is that it's not quite clear how a script would know that the dsp_power_state has been reached.
How long should the script wait and does it get any error codes?

This seems to simple to be useful IMHO.

Reply:
@plbossart i have exposed both the current fw boot state and the dsp power state that the script can poll to see if the desired state has been reached. How long to poll something that we can set based on the current kernel poll wait times. As for error codes, Im afraid I have no way to pass that on to userspace but we have everything we need in the dmesg isnt it?